packages/flutter/lib/src/widgets/interactive_viewer.dart

```diff
@@ -660,7 +660,8 @@ class _InteractiveViewerState extends State<InteractiveViewer> with TickerProvid
       _viewport.width / _boundaryRect.width,
       _viewport.height / _boundaryRect.height,
     );
```

Comment:
Is `minScale` the max scale allowed?

if `_boundaryRect.size` `(10, 10)` and the viewport's size is `(100, 200)`, then `minScale` is going to be 20 right? Anything above 20 is going to make the child exceed the size of the viewport in both dimensions.

Reply:
This minScale is the minimum scale allowed, but maybe my comment in the code above this is confusing.

The boundary is defined as being positioned on the child, and it scales with the child.  In your example of a 10,10 boundary, let's say that the child is also of size 10,10.  So here's your example with an orange child and a yellow viewport:

| scale | 1.0 | 10.0 | 20.0 |
| -- | -- | -- | -- |
|           | <img width="151" alt="Screen Shot 2020-09-10 at 9 37 20 AM" src="https://user-images.githubusercontent.com/389558/92764802-9d926280-f349-11ea-8ca2-85fde7e1627d.png"> | <img width="173" alt="Screen Shot 2020-09-10 at 9 37 02 AM" src="https://user-images.githubusercontent.com/389558/92766646-41304280-f34b-11ea-879e-682a18e82b9d.png"> | <img width="196" alt="Screen Shot 2020-09-10 at 9 37 40 AM" src="https://user-images.githubusercontent.com/389558/92765015-cfa3c480-f349-11ea-9fae-363d0987785e.png"> |

The purpose of this minScale is to prevent the viewport from exposing anything beyond the boundary. In other words, in this example it's trying to cover up all the yellow using the orange child.  It is not trying to prevent the child from growing beyond the size of the viewport, that's fine.

Reply:
Ah I see. Thanks for the explanation! In that case would it be possible to move the logic in the if clause to before clamping? Then the function would be roughly
```dart
    final double currentScale = _transformationController.value.getMaxScaleOnAxis();
    final double totalScale = math.max(currentScale * scale, math.max(_viewport.width / _boundaryRect.width, _viewport.height / _boundaryRect.height));  
    final double clampedTotalScale = totalScale.clamp(widget.minScale, widget.maxScale);
    final double clampedScale = clampedTotalScale / currentScale;
    return matrix.clone()..scale(clampedScale);
```

Reply:
Yes that looks much better, thanks!